### PR TITLE
merger_config: add lists of conflicts

### PR DESCRIPTION
**Pull request by rikirenz**
* Adds the lists of conflicts that we have to filter and show

Signed-off-by: Riccardo Candido <riccardo.candido@cern.ch>

### CODE REVIEW

File: inspire_json_merger/merger_config.py [hunk 1]
```diff
@@ -80,11 +80,22 @@ def factory(config_type):
...
 class MergerConfigurationOperations(object):
+    """General Configuration for the Merger
+
+    Attributes:
+        default_dict_merge_op   Default operation for the merge of dicts
+        default_list_merge_op   Default operation for the merge of lists
+        list_dict_ops = None    Dictionary of rules for the merge of dicts
+        list_merge_ops = None   Dictionary of rules for the merge of dicts
+        comparators = None      Dictionary of comparators
+        conflicts = None        List of conflicts that have to be filtered
```
Comment (ammirate) [inline inspire_json_merger/merger_config.py]: Maybe we can improve this docstring using the Napoleon style: `var_name(var_type): <<description>>` :books:

File: inspire_json_merger/merger_config.py [hunk 2]
```diff
@@ -80,11 +80,22 @@ def factory(config_type):
...
+    """
     default_dict_merge_op = DictMergerOps.FALLBACK_KEEP_UPDATE
     default_list_merge_op = UnifierOps.KEEP_ONLY_UPDATE_ENTITIES
     list_dict_ops = None
     list_merge_ops = None
     comparators = None
+    conflicts = None
```
Comment (ammirate) [inline inspire_json_merger/merger_config.py]: The name of this var is not so self explaining...what do you think about something like `relevant_conflicts` or `conflicts_to_show`?

File: inspire_json_merger/merger_config.py [hunk 3]
```diff
@@ -425,3 +475,17 @@ class PublisherToPublisherOperations(MergerConfigurationOperations):
         'title_translations': DictMergerOps.FALLBACK_KEEP_HEAD,
         'urls': DictMergerOps.FALLBACK_KEEP_HEAD,
     }
+    conflicts = [
```
Comment (ammirate) [inline inspire_json_merger/merger_config.py]: same of above :arrow_up:

File: inspire_json_merger/merger_config.py [hunk 4]
```diff
@@ -314,6 +340,30 @@ class PublisherToArxivOperations(MergerConfigurationOperations):
         'title_translations': DictMergerOps.FALLBACK_KEEP_HEAD,
         'urls': DictMergerOps.FALLBACK_KEEP_HEAD
     }
+    conflicts = [
+        '_collections',
```
Comment (ammirate) [inline inspire_json_merger/merger_config.py]: same of above :arrow_up:

File: inspire_json_merger/merger_config.py [hunk 5]
```diff
@@ -196,6 +207,22 @@ class ArxivToArxivOperations(MergerConfigurationOperations):
         'title_translations': DictMergerOps.FALLBACK_KEEP_HEAD,
         'urls': DictMergerOps.FALLBACK_KEEP_HEAD,
     }
+    conflicts = [
+        '$schema',
```
Comment (ammirate) [inline inspire_json_merger/merger_config.py]: same of above :arrow_up:

File: inspire_json_merger/merger_config.py (hunk 2 above)
Comment (rikirenz) [inline inspire_json_merger/merger_config.py]: I really like `conflicts_to_show` :smile:

File: inspire_json_merger/merger_config.py (hunk 1 above)
Comment (rikirenz) [inline inspire_json_merger/merger_config.py]: you are definitely right! could you give me an example? :smile:

**Pull request merged**